### ISSUE TITLE
Unparsable bed record error while load indexed BED file track

### ISSUE THREAD

**Issue by RoadRoller** [igvteam/igv.js#1314]
igv.js version is 2.7.4.

I added custom BED file track with following config:

``` javascript
{
      name: "Splicing sites",
      url: `<MY_SERVER_URL>/splice-junction.bed.gz`,
      indexURL: `<MY_SERVER_URL>/splice-junction.bed.gz.tbi`,
      type: "annotation",
      format: "bed",
      visibilityWindow: 10000
}
```
Source and index files prepared in a standard way with bgzip and tabix utils.

For the most part of regions it works as expected, but for some regions I face with errors like "Unparsable bed record: chr2 208271304".

After some investigations I found out that when IGV request part of BED, response ends in the middle of the line:
```
...
chr2    208271275    208271277
chr2    208271277    208271279
chr2    208271284    208271286
chr2    208271289    208271291
chr2    208271296    208271298
chr2    208271301    208271303
chr2    208271304    
```
last row ends with tab.

So exception throws from line 25 of file  [js/feature/decode/ucsc.js](https://github.com/igvteam/igv.js/blob/9483c0aad6073c7a97d8fb6dc26d35cc017e88cf/js/feature/decode/ucsc.js):

```
    const end = tokens.length > 2 ? parseInt(tokens[2]) : start + 1;
    if (isNaN(start) || isNaN(end)) {
        throw Error(`Unparsable bed record: ${tokens.join('\t')}`);
    }
```
because of end here is empty string.

Сould it be related to wrong preparation of gz/tbi?

Or if it's bug, I ready to create pull request, but not understand what the original reason: is it expected when we request part of indexed file that start/end of part can be obtained in random locations or they should be aligned to start/end of line?

If no, as I see method `decodeBed(tokens, header)` that throws exception tolerant to input data, for example it just returns undefined in case when tokens length less than three. Maybe in this case when start or end is NaN, it also should returns undefined instead of throws exception?

**Comment by northwestwitch** [comment]
What version of IGV.js are you using? Our software has a [similar problem](https://github.com/Clinical-Genomics/scout/issues/2565). We are using version 2.7.5

**Comment by northwestwitch** [comment]
Tested version 2.8 now and we still see the problem. Our issue arises in genes track, when you look for a specific gene (FLT3). This is the gene track we use: https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg19/genes/refGene.hg19.bed.gz

**Comment by jrobinso** [comment]
This is fixed now and released in 2.8.1
